app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/SignInActivityTest.kt

```diff
@@ -19,6 +21,14 @@ class SignInActivityTest {
     @get:Rule
     val activityRule = ActivityScenarioRule(SignInActivity::class.java)
 
+    @Test
+    fun testSignInWithCurrentUser() {
+        // Simulate a signed-in user
+        FirebaseAuth.getInstance().signInAnonymously().addOnSuccessListener {
+            // Perform a click on the button that triggers the code snippet
+            onView(withId(R.id.signInGoogleLayout)).perform(click())
+        }
+    }
```

Comment:
If you apply my proposed changes below, we could now mock the `FirebaseAuth` in this test file to test more in depts!